runtime/Java/src/org/antlr/v4/runtime/tree/SyntaxTree.java

```diff
@@ -42,10 +42,14 @@
 	 * Return an {@link Interval} indicating the index in the
 	 * {@link TokenStream} of the first and last token associated with this
 	 * subtree. If this node is a leaf, then the interval represents a single
-	 * token.
+	 * token and has interval i..i for token index i.
+	 *
+	 * <p>An interval of i..j for j &lt; i indicates an empty interval.</p>
```

Comment:
The `getSourceInterval()` method was updated in 81e2a654d772b630933d98ee6a8c1fe08a7d402a to ensure that an empty interval is always represented as i..i-1. I recommend changing this paragraph to the following:

```
<p>An interval of i..i-1 indicates an empty interval at position i in the input stream,
where 0 &lt;= i &lt;= the size of the input token stream.</p>
```